CliFx/Services/EnvironmentVariablesProvider.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security;
+
+namespace CliFx.Services
+{
+    /// <inheritdoc />
+    public class EnvironmentVariablesProvider : IEnvironmentVariablesProvider
+    {
+        /// <summary>
+        /// Default delimiter for multiple values
+        /// </summary>
+        public const char DEFAULT_VALUES_DELIMITER = ',';
+
+        private readonly char _valuesDelimiter;
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="EnvironmentVariablesProvider"/> with the values delimiter specified
+        /// </summary>
+        public EnvironmentVariablesProvider(char valuesDelimiter)
+        {
+            _valuesDelimiter = valuesDelimiter;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="EnvironmentVariablesProvider"/> with the default values delimiter
+        /// </summary>
+        public EnvironmentVariablesProvider()
+            : this(DEFAULT_VALUES_DELIMITER)
+        {
+        }
+
+        /// <inheritdoc />
+        public IReadOnlyList<string> GetValues(string variableName)
+        {
+            //If variableName is null simply return nothing, this may happen if a Command has not EnvironmentVariable fallback
+            if (variableName == null) return null;
+
+            try
+            {
+                string variableValue = Environment.GetEnvironmentVariable(variableName);
+
+                if (string.IsNullOrWhiteSpace(variableValue)) return null;
+
+                var values = variableValue.Split(_valuesDelimiter)
+                    .Select(v => v.Trim());
+
+                return values.ToList();
+            }
+            catch (SecurityException)
```

Comment:
When does this happen, out of curiosity?

Reply:
According to Microsoft documentation it says that 

> The caller does not have the required permission to perform this operation.

I believe this might happen in Windows if an user is under some group policy that forbids reading some env. variables although that's speculation but I tought that given that the documentation reports it it is always better to handle it even if by simply swallowing it and returning `null`.

Reply:
Hm. It will be hard to cover this. Maybe throwing an exception makes sense, the user would want to know why environment variable is not resolved. It will only be called if `EnvironmentVariableName` is set to it shouldn't interfere with other options.